NFC Reader Feature

NxFerrara opened the pull request:
## Changes:
- Added NFC Reader module

## Considerations:
- The module is made by making a python wrapper for the c implementation
- We will no longer be using nfcpy since there seems to be incompatibility issues

NxFerrara committed:
Fix spaces around var assignment

The shell script interprets the variable names as commands due to the spacing around the equals sign.

NxFerrara committed:
Fix function name reference

NxFerrara committed:
Change apt -> apt-get in 3_create_venv.sh

apt-get is meant for the CLI in scripting while apt is a simplified version that is meant to be user-friendly.

NxFerrara committed:
Fix environment var name reference

NxFerrara committed:
Fix bad service unit file

The working directory cannot be in quotes otherwise it gives an error.

NxFerrara committed:
Fix empty device state file

There needs to be a default of {} in the file otherwise you cannot read the file as JSON.

NxFerrara committed:
Fix empty device state case when reading

Reading from an empty dictionary gave an error since the device state dictionary had no 'saved_timestamp' key.

NxFerrara committed:
Add popup limit for invalid taps

There is an issue with the invalid taps having many popups queued, which essentially makes the display inoperable until the queue is empty again. The popup limit aims to avoid these situations.

NxFerrara committed:
Add basic nfcpy test

NxFerrara committed:
Update nfc_reader.py

NxFerrara committed:
Fix time comparison error

The time was being compared lexicographically, which may have caused issues.

NxFerrara committed:
Fix time comparison error

The time was being compared lexicographically, which may have caused issues.

NxFerrara committed:
Fix config variable declaration

Treated the variables as commands instead.

NxFerrara committed:
Fix variable substitution with braces

NxFerrara committed:
Fix variable substitution with braces

NxFerrara committed:
Add c program for NFC polling

This is a first iteration to just test the NFC reader.

NxFerrara committed:
Add time release function of SelfReleasingLock

NxFerrara committed:
Fix types, comments, and style

NxFerrara committed:
Fix global variable reference

NxFerrara committed:
Modularize read_device_state

The resetting of the unit count should happen independently from the reading of the device state for readability, robustness, etc.

NxFerrara committed:
Remove unnecessary comments

NxFerrara committed:
Make JSONDecodeError message clearer

NxFerrara committed:
Add Makefile for easy compilation

NxFerrara committed:
Fix source directory reference in Makefile

NxFerrara committed:
Merge branch 'bugfix/shutdown' into feature/nfc-reader

NxFerrara committed:
Fix read/write device state logic

The fix was to make the code more readable, so the read_device_state function should solely read the device state, and nothing more. Same applies to write_device_state.

NxFerrara committed:
Rename nfc_reader.c -> poll.c

NxFerrara committed:
Create wrapper_test.py

NxFerrara committed:
Update wrapper_test.py

kc34251 committed:
NFC_reader initial outline

NxFerrara committed:
Fix dereferencing UID pointer

NxFerrara committed:
Fix type of UID

The unsigned integer should be converted from hex to a string.

NxFerrara committed:
Fix return type (char[] -> char*)

NxFerrara committed:
Fix return values

NxFerrara committed:
Add buffer argument to poll

Since we can't easily return a char array, we instead need to either use a dynamic array that will need to be freed by the caller at some point or simply modify a buffer passed by the caller.

NxFerrara committed:
Change uint_to_hexstr return type to bool

The conditional check of whether the function was a success or failure is needed, so this change was made.

NxFerrara committed:
Fix cleanup order

NxFerrara committed:
Make polling indefinite

The poll function should poll until it finds an NFC tag.

NxFerrara committed:
Remove main (unused)

NxFerrara committed:
Update main to call poll

NxFerrara committed:
Merge branch 'feature/nfc-reader' of https://github.com/NxFerrara/FiberPTS into feature/nfc-reader

NxFerrara committed:
Add debug statements to uint_to_hexstr

NxFerrara committed:
Add missing semi-colon

NxFerrara committed:
Add TODO

NxFerrara committed:
Add Invalid Argument(s) explanation

NxFerrara committed:
Comment the stable gcc command

NxFerrara committed:
Add TODO for makefile

kc34251 commented on src/nfc_reader/poll.c:
```diff
@@ -0,0 +1,129 @@
+// (i.e. removes the .o files except for the libnfc ones)
+
+// Function to handle the stop signal (SIGINT)
+static void stop_polling(int sig)
```
What is the stop polling function for? Is this in case of shutdown or power loss?

NxFerrara committed:
Fix uint_to_hexstr potential error

NxFerrara committed:
Fix seg fault with uint_to_hexstr

NxFerrara committed:
Add debug statement

NxFerrara committed:
Fix uninitialized UID length

NxFerrara committed:
Switch to single target polling

NxFerrara committed:
Add missing library

NxFerrara committed:
Fix shared library name reference

NxFerrara committed:
Make waiting before polling

The previous tag should not be on the reader still.

NxFerrara committed:
Fix Makefile (nfc-utils.c not needed)

NxFerrara committed:
Refactor (comments and style)

NxFerrara committed:
Add is_tag_present

Need a function to check if there is a tag present

NxFerrara committed:
Add statement to check tag presence

NxFerrara committed:
Fix detection logic

NxFerrara committed:
Add wait tag removal logic

NxFerrara committed:
Add is_tag_present function description

NxFerrara committed:
Add TODO for signal handler

The signal handler does not operate when called from python

kc34251 committed:
Version 2 of nfc_reader outline

NxFerrara committed:
Change NFC Reader pipe variable name

NxFerrara committed:
Add handling nfc pipe data

NxFerrara committed:
Add NFC data to dashboard

kc34251 committed:
Initial outline of lookup_uid function in cloud_db. Needs work to figure out how the tables will look in Supabase when synced up with Airtable

kc34251 committed:
Minor improvements to code format and variable naming

NxFerrara committed:
Merge branch 'bugfix/setup' into feature/nfc-screen

NxFerrara committed:
Minor fixes

NxFerrara committed:
Make read/write device state handled by touch_sensor

NxFerrara committed:
Merge branch 'bugfix/setup' into feature/nfc-screen

NxFerrara committed:
Merge branch 'refactor/nxferrara' into feature/nfc-screen

NxFerrara committed:
Minor fixes with global variables & style

NxFerrara committed:
Revert "Make read/write device state handled by touch_sensor"

 This reverts commit 419ae203396aaf14939d3a6617054519d6e2ff50.

NxFerrara committed:
Add device state write/read handling for screen.py

kc34251 committed:
Changed NFCTag, NFCType. Additionally added functionality to insert employee taps and order taps into cloud_db.py

NxFerrara committed:
Merge branch 'feature/nfc-reader' into feature/nfc-screen

NxFerrara committed:
Merge remote-tracking branch 'origin/feature/nfc-cloud-db' into feature/nfc-screen

NxFerrara committed:
Resolve minor merge conflict

NxFerrara committed:
Remove nfcpy from requirements.txt

NxFerrara committed:
Create test_nfc_tag.py

NxFerrara committed:
Add Invalid NFC Tap

kc34251 committed:
Added full insert employee and order tap functionality

NxFerrara committed:
Fix assignment spacing

kc34251 committed:
Change insert_order_tap to accomodate missing employee_id

NxFerrara committed:
Merge remote-tracking branch 'origin/feature/nfc-cloud-db' into feature/nfc-python-module

NxFerrara committed:
Fix lookup_uid function reference

NxFerrara committed:
Fix insert employee/order and lookup uid

NxFerrara committed:
Merge branch 'feature/nfc-python-module' into feature/nfc-reader

NxFerrara committed:
Add NFC logic to insert tap

NxFerrara committed:
Add null checks for data values

Some data like machine_id can be null, but to be careful, we want to only populate the record with either a value or nothing where nothing is interpreted as null.

NxFerrara committed:
Fix line offset reference

NxFerrara committed:
Fix incorrect function import

NxFerrara committed:
Make pipe reading non-blocking

We need this so that the screen can read sequentially from nfc_reader and touch_sensor pipes.
